### PR TITLE
enable diff norm types

### PR DESCRIPTION
**Pull request by vchiley**
pass norm type and add rmsnorm as an option

issue: https://mosaicml.atlassian.net/browse/RESEARCH-572

### CODE REVIEW

File: examples/llm/src/models/layers/norm.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+import torch
+from composer.algorithms.low_precision_layernorm.low_precision_layernorm import \
```
Comment (dakinggg) [inline examples/llm/src/models/layers/norm.py:4]: can you copy lpln from composer into examples as part of this pr? it makes it easier to remove the composer dependency for hf release

File: examples/llm/src/models/layers/norm.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+NORM_CLASS_REGISTRY = {
+    'layernorm': torch.nn.LayerNorm,
+    'ln': torch.nn.LayerNorm,  # alias
+    'low_precision_layernorm': LPLayerNorm,
+    'lplayernorm': LPLayerNorm,  # alias
+    'lpln': LPLayerNorm,  # alias
+    'rmsorm': RMSNorm,
+    'low_precision_rmsorm': LPRMSNorm,
+    'lprmsorm': LPRMSNorm,  # alias
```
Comment (abhi-mosaic) [inline examples/llm/src/models/layers/norm.py:117]: will having aliases make it confusing for people if they save two models with configs who differ by an alias, but are really using the same norm_type?

**Review by dakinggg — COMMENTED** [review]
Add a unit test for constructing the diff norm types please :)

File: examples/llm/src/models/layers/norm.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+NORM_CLASS_REGISTRY = {
+    'layernorm': nn.LayerNorm,
+    'ln': nn.LayerNorm,  # alias
+    'low_precision_layernorm': LPLayerNorm,
+    'lplayernorm': LPLayerNorm,  # alias
+    'lpln': LPLayerNorm,  # alias
+    'rmsorm': RMSNorm,
+    'low_precision_rmsorm': LPRMSNorm,
+    'lprmsorm': LPRMSNorm,  # alias
```
[the remark this answers is not in the document]
Comment (vchiley) [reply examples/llm/src/models/layers/norm.py:117]: per abhi's suggestion, going to rm alias versions.

File: examples/llm/src/models/mosaic_gpt/configuration_mosaic_gpt.py [hunk 4]
```diff
@@ -44,7 +44,7 @@ def __init__(
         fan_mode: str = 'fan_in',
         init_nonlinearity: str = 'relu',
         embedding_fraction: float = 1.0,
-        low_precision_layernorm: bool = True,
+        norm_type: str = 'low_precision_layernorm',
```
Comment (dakinggg) [inline examples/llm/src/models/mosaic_gpt/configuration_mosaic_gpt.py:47]: should we accept the old arg and just map it to the correct norm type for backwards compat reasons?

Comment (vchiley) [reply examples/llm/src/models/mosaic_gpt/configuration_mosaic_gpt.py:47]: what you're describing is just a config updt; I wouldn't worry about it.


Changing this
```
        self.ln_1 = layernorm_class(d_model, device=device)
```
to this
```
        self.norm_1 = norm_class(d_model, device=device)
```
is the bigger bwd compat issue.
(param names from old models will be diff; cant just load them)

Comment (dakinggg) [reply examples/llm/src/models/mosaic_gpt/configuration_mosaic_gpt.py:47]: that is fair

**Review by dakinggg — APPROVED** [review]
lgtm

**Pull request merged**